src/components/TextInput/TextInput.js

```diff
@@ -123,7 +126,13 @@ function PrivateTextInput({
     } else {
       returnClasses = `${styles.TextInputCommon} ${styles.TextInputStylable}`
     }
-
+    // hasIcon class indicates the text input has icon, to give more padding-right in stylings,to prevent overlapping between icon and long input
+    if (
+      (iconPrefix === 'fas' && iconName === 'lock') ||
+      (iconPrefix === 'far' && iconName === 'eye-slash')
```

Comment:
Is there a reason we're limiting users to only using these combinations of icons? In the future if someone wants a different icon, would they have to change this code? I would prefer we don't limit this now, so no changes are needed later to use different icons.

Reply:
Resolving since we discussed on zoom.